code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -114,6 +114,27 @@
 	ammo_type = /obj/item/ammo_casing/shotgun/improvised
 	max_ammo = 1
 
+/obj/item/ammo_box/magazine/internal/shot/improvised/New()
+ 	update_icon()
+
+/obj/item/ammo_box/magazine/internal/shot/improvised/i45
+	name = "pistol carbine internal magazine"
+	ammo_type = /obj/item/ammo_casing/c45
+	caliber = ".45"
+	max_ammo = 1
```

Comment:
Use plasteel for magazine extensions then. Maybe rods too but I can't think of anything extra aside from those